### ISSUE TITLE
Not showing all comments

### ISSUE THREAD

**Issue by Donkey-Doug** [Tunous/Dawn#236]
![Screenshot_2020-06-11-00-36-57-743_me thanel dank](https://user-images.githubusercontent.com/50382414/84325723-b426f300-ab7b-11ea-807a-c0a97769a2b9.jpg)
**Describe the bug**
For some posts (very few) it shows the correct number of comments (below the thumbnail) prior to opening the post. However, after I try ro read the comments I notice the number of comments is a lot less than shown. I opened the mobile site of reddit in my browser. Turns out the number of comments is displayed correctly, but not all comments are shown.

**To Reproduce**
1. Go to r/android in dawn
2. Search for the post with title "Do you remember Dank for Reddit? Check out Dawn, a fork that is being updated"
3. Take note of the number of comments shown below the thumbnail.
4. Open the post and count the number of comments. The amount of comments is less than shown below the thumbnail.
5. Open the same post in for examole your browser and notice that there are comments missing in Dawn.

**Smartphone:**
 - Device: Xiaomi Mi 9T Pro
 - OS: Android 10 (MIUI 11)
 - Version: 0.9.1

**Comments:**
In the attachted screenshot all the comments are visible. Notice it is way less than the claimed 33 comments.

**Comment by AbsurdlySuspicious** [comment]
This is not a Dawn bug, Reddit api returns incorrect comment count. There is also no real way to count them client-side in most cases since comments are loaded on demand while scrolling. The only possible workaround for this is to set correct comment count after acquiring last offset

**Comment by AbsurdlySuspicious** [comment]
https://old.reddit.com/r/Android/comments/gzr5aj/do_you_remember_dank_for_reddit_check_out_dawn_a/ - thread link

Situation with incorrect number of comments indeed happens pretty frequently, but I haven't ever seen missing comments compared to web version. This one too - count of comments is exactly same as counter says in both app and web (33 at the moment of writing this). A more reproducible example would be nice so we can confirm that's indeed api issue

**Comment by Donkey-Doug** [comment]
> This is not a Dawn bug, Reddit api returns incorrect comment count.

This is not true. If you open the same post in a browser you will see more comments.

**Comment by Donkey-Doug** [comment]
> 
> 
> https://old.reddit.com/r/Android/comments/gzr5aj/do_you_remember_dank_for_reddit_check_out_dawn_a/ - thread link
> 
> Situation with incorrect number of comments indeed happens pretty frequently, but I haven't ever seen missing comments compared to web version. This one too - count of comments is exactly same as counter says in both app and web (33 at the moment of writing this). A more reproducible example would be nice so we can confirm that's indeed api issue

Again what you say is completely incorrect. If you open reddit in the browser of your laptop/PC, you see 33 comments. If you open the mobile site of reddit on your phone, you see 33 comments. If you download another reddit app from the playstore (I tried Boost for reddit), you see 33 comments. It is a Dawn bug.

**Comment by AbsurdlySuspicious** [comment]
Have you tried to press refresh? Dawn doesn't update comments automatically

**Comment by Donkey-Doug** [comment]
Refreshing makes no difference. Still missing comments.

**Comment by AbsurdlySuspicious** [comment]
Seems that I completely misinterpreted the initial cause of this. That's really strange and looks like there's no easy way to reproduce this. Can you provide logcat logs after you hit refresh in this thread?

**Comment by Donkey-Doug** [comment]
Well, I just manually refreshed and it did capture all comments. I guess the app does not check for new comments if a post that has previously already been opened gets opened again.

Also, how do I provide logs?

**Comment by AbsurdlySuspicious** [comment]
That's expected behavior for now. We'll need to address it of course and make it more user friendly. I'll drop issue number on this topic here

As for logs, in future you can use 
Logcat Reader (A simple app to view logs) - https://f-droid.org/app/com.dp.logcatapp
Just start it before application, press clear in 3-dot menu and then, after reproducing the bug, use save option in 3-dot menu.
